debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -385,18 +446,13 @@ private boolean useFailOverSlot() throws SQLException {
             return false;
         }
 
-        AtomicBoolean isPrimary = new AtomicBoolean();
-        query("select pg_is_in_recovery()", rs -> {
-            if (rs.next()) {
-                isPrimary.set(!rs.getBoolean(1));
-            }
-        });
+        boolean isPrimary = isReadOnlyDb();
```

Comment:
Could you please either rename the variable or change to
```suggestion
        boolean isPrimary = !isReadOnlyDb();
```
and update the following condition? The code would be a bit clearer.